core/src/impl/Kokkos_Core.cpp

```diff
@@ -711,7 +756,9 @@ void parse_environment_variables(InitArguments& arguments) {
     for (char& c : env_str) {
       c = toupper(c);
     }
-    if ((env_str == "TRUE") || (env_str == "ON") || (env_str == "1"))
+    const auto _rc = std::regex_constants::icase | std::regex_constants::egrep;
+    const auto _re = std::regex("^(true|on|yes|[1-9])$", _rc);
+    if (std::regex_match(env_str, _re))
```

Comment:
I don't necessarily have an issue with it but this is a change in behavior so you must add it to the description of the PR.

Reply:
Ah right